modules/imgproc/src/filter.simd.hpp

```diff
@@ -465,6 +465,72 @@ struct RowVec_8u32s
     bool smallValues;
 };
 
+struct RowVec_8u32f
+{
+    RowVec_8u32f() {}
+    RowVec_8u32f( const Mat& _kernel ) { kernel = _kernel; }
+
+    int operator()(const uchar* _src, uchar* _dst, int width, int cn) const
+    {
+        CV_INSTRUMENT_REGION();
+
+        int ksize = kernel.rows + kernel.cols - 1;
+        const float* kx = kernel.ptr<float>();
+        const uchar* S;
+        float* D = (float*)_dst;
+        int i, k;
+
+        i = 0;
+        width *= cn;
+        for( ; i < width - v_uint8::nlanes; i += v_uint8::nlanes )
+        {
+            S = (const uchar*)_src + i;
+            v_uint8 vs = vx_load(S);
+            v_float32 vkx = vx_setall_f32(kx[0]);
```

Comment:
Is there a particular reason to do the 0th index work, then the rest separately? Would there be any downsides to removing the 0th index, initialising `s0_ll` etc to all zeros and then doing
```
for( k = 0; k < ksize; k++ )
``` 
on line 504?